Add startup tuning documentation (#2455)
elastic/apm-agent-java: docs/tuning-and-overhead.asciidoc

This is where it's important to know how many requests your application handles and how many of those you want to record and store.
This can be adjusted with the <<tune-sample-rate>>.

[float]
[[tuning-agent-startup]]
=== Tuning the Agent Startup

When the Java agent starts, it needs to initialize various components of the agent, connect
to the APM server, and instrument any already loaded classes that it has been configured to

> trace. This takes some time and resources, and if done synchronously from JVM start (which is

Reviewer: [minor]
```suggestion
trace. This takes some time and resources, and if done synchronously on the main thread (which is
```
